syntax = "proto3";

package math;

service MathService {
  rpc Add (Parm2Request) returns (Parm2Result);
  rpc Sub (Parm2Request) returns (Parm2Result);
  rpc Mul (Parm2Request) returns (Parm2Result);
  rpc Div (Parm2Request) returns (Parm2Result);
  rpc Pow (Parm2Request) returns (Parm2Result);
}

message Parm2Request {
  double x = 1;
  double y = 2;
}

message Parm2Result {
  double z = 1;
  string error = 2;
}